plasmapy/diagnostics/tests/test_path_integrated_diagnostic.py

```diff
@@ -0,0 +1,173 @@
+
+    """
+    plt.plot(hax, line)
+    plt.plot(hax, theory)
+    plt.show()
+    """
+    assert np.allclose(line, theory, atol=2e-4)
+
+
+def test_non_collimated(grid):
+    """
+    Tests line-integration with a point source by checking that the
+    width of the feature scales with the magnification.
+    """
+    source = (0 * u.mm, -5 * u.mm, 0 * u.mm)
+    detector = (0 * u.mm, 10 * u.mm, 0 * u.mm)
+    obj = LineIntegrateScalarQuantities(grid, source, detector, "rho", verbose=False)
+
+    # Run the same test but with a point source (not collimated)
+
+    size = np.array([[-1, 1], [-1, 1]]) * 5 * u.mm
+    hax, vax, integral = obj._line_integral(
+        size=size, bins=[100, 100], collimated=False, num=100
+    )
+
+    # Scaling hax/mag gives hax in the object plane
+    # That will now agree with the theory
+    hax = hax.to(u.mm).value / obj.mag
+    vax = vax.to(u.mm).value
+
+    integral = integral.to(u.kg / u.m ** 2).value
+    line = np.mean(integral, axis=1)
+
+    theory = np.where(np.abs(hax) < 0.999, 2 * np.sqrt(np.abs(1 - hax ** 2)) * 1e-3, 0)
+
+    """
+    import matplotlib.pyplot as plt
+    plt.plot(hax, line)
+    plt.plot(hax, theory)
+    plt.show()
+    """
+
+    assert np.allclose(line, theory, atol=5e-4)
+
+
+def test_constant_box(grid):
+    x, y, z = grid.grid.T
+    t1 = np.where(np.abs(x) < 1 * u.mm, 1, 0)
+    t2 = np.where(np.abs(z) < 1 * u.mm, 1, 0)
+    field = np.where(t1 * t2 != 0, 1, 0) * u.kg / u.m ** 3
+    grid.add_quantities(rho=field)
+
+    source = (0 * u.mm, -5 * u.mm, 0 * u.mm)
+    detector = (0 * u.mm, 5 * u.mm, 0 * u.mm)
+    obj = LineIntegrateScalarQuantities(grid, source, detector, "rho", verbose=False)
+
+    # Test that line-integrating with collimated = False yields
+
+    size = np.array([[-2, 2], [-2, 2]]) * u.mm
+    hax, vax, integral = obj._line_integral(
+        size=size, bins=[100, 100], collimated=True, num=100
+    )
+    hax = hax.to(u.mm).value
+    vax = vax.to(u.mm).value
+
+    line = np.mean(integral, axis=1)
+
+    # The value should be the density in the rectangular region times
+    # it's width, which is 2 mm
+    val = (1 * u.kg / u.m ** 3 * 2 * u.mm).to(integral.unit)
+    theory = np.where(np.abs(hax) < 1, val, 0 * integral.unit)
+
+    """
+    plt.plot(hax, line)
+    plt.plot(hax, theory)
+    plt.show()
+    """
+
+    assert np.allclose(line, theory, atol=2e-4)
+
+
+def test_interferogram_sphere(grid):
+    x, y, z = grid.grid.T
+    r = np.sqrt(x ** 2 + y ** 2 + z ** 2)
+    n_e = np.where(r < 1 * u.mm, 1, 0) * 3e19 / u.cm ** 3
+    grid.add_quantities(n_e=n_e)
+
+    source = (0 * u.mm, -5 * u.mm, 0 * u.mm)
+    detector = (0 * u.mm, 5 * u.mm, 0 * u.mm)
+    obj = Interferometer(grid, source, detector, verbose=False)
+
+    size = np.array([[-2, 2], [-2, 2]]) * u.mm
+    bins = [350, 350]
+
+    hax, vax, phase = obj.evaluate(
+        1.14e15 * u.Hz,
+        size=size,
+        bins=bins,
+        num=100,
+        unwrapped=False,
+    )
```

Comment:
Just checking that this evaluates?

Would it help if we had mechanisms in place to verify that our tests keep returning the same image as they did previously? I'm asking because I've been putting #779 off for a while (started dabbling it a little while ~~frustrated with~~ working on https://github.com/PlasmaPy/PlasmaPy/pull/1059 ), and I could get back to it if it made this kind of test easier.

Reply:
Yes I would use that all the time! Ideally it would include some sort of `np.isclose` rtol keyword or something. As is I've been writing custom checks based on the geometry of each image and I just haven't thought of one for this one yet.

Reply:
All right, I'll try to work on that this weekend, then :)